src/System.Data.Common/src/System/Data/Common/DbProviderFactory.netcoreapp.cs

```diff
@@ -0,0 +1,18 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Data.Common
+{
+    public abstract partial class DbProviderFactory
+    {
+        private bool? _canCreateDataAdapter;
+        private bool? _canCreateCommandBuilder;
+
+        public virtual bool CanCreateDataAdapter
+            => (_canCreateDataAdapter ??= CreateDataAdapter() != null).Value;
+
+        public virtual bool CanCreateCommandBuilder
+            => (_canCreateCommandBuilder ??= CreateCommandBuilder() != null).Value;
```

Comment:
Do we need to dispose of the resulting command builder object?

Reply:
I think these are disposable only because they inherit from Component - hard to imagine a case where they'd actually hold any resources - but I'll do it just in case, thanks!

Reply:
BTW no CI??